src/plugin.js

```diff
@@ -3,32 +3,58 @@
  * The main file for the HLS project.
  * License: https://github.com/videojs/videojs-contrib-hls/blob/master/LICENSE
  */
-(function(window, videojs, document, undefined) {
-'use strict';
 
-var
-  // A fudge factor to apply to advertised playlist bitrates to account for
-  // temporary flucations in client bandwidth
-  bandwidthVariance = 1.2,
-  blacklistDuration = 5 * 60 * 1000, // 5 minute blacklist
-  TIME_FUDGE_FACTOR = 1 / 30, // Fudge factor to account for TimeRanges rounding
-  Component = videojs.getComponent('Component'),
+import {MediaSource, URL} from 'videojs-contrib-media-sources';
+import PlaylistLoader from './playlist-loader';
+import Playlist from './playlist';
+import m3u8 from './m3u8';
+import {Decrypter, AsyncStream, decrypt} from './decrypter';
+import utils from './bin-utils';
+import xhr from './xhr';
+import resolveUrl from './resolve-url';
+import videojs from 'video.js';
+
+const Player = videojs.getComponent('Player');
+
+if (typeof Player.prototype.MediaSource === 'undefined') {
+  videojs.plugin('MediaSource', MediaSource);
+  videojs.plugin('URL', URL);
```

Comment:
These objects don't make sense as plugins to me. Plugins are intended for player-level customizations. MediaSource and URL don't affect individual player operation.